### PR TITLE
Kasten K10 pattern

### PR DESCRIPTION
**Pull request by ajbergh**
Kasten K10 Pattern Construct

*Description of changes:*

Adds Kasten K10 Pattern Construct to /lib/kasten-k10-construct

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/kasten-k10-construct/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { App } from 'aws-cdk-lib';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import { KastenK10AddOn } from '@kastenhq/kasten-eks-blueprints-addon';
+
+const app = new App();
+
+blueprints.EksBlueprint.builder()
+    .addOns(new blueprints.ClusterAutoScalerAddOn)
```
Comment (youngjeong46) [inline lib/kasten-k10-construct/index.ts:8]: Is Cluster Autoscaler AddOn necessary for Kasten to operate?

**Review by youngjeong46 — COMMENTED** [review]
Hi @ajbergh,

Thanks for the PR. Patterns should leverage a construct that can be initialized at entrypoint (bin/main.ts). I can do a separate patterns PR (drafted here: #27) once you can confirm the Cluster Autoscaler is necessary for the addon to operate (and close this one). Upon my own testing I don't think it does, but just wanted to confirm.

File: lib/kasten-k10-construct/index.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    .addOns(new KastenK10AddOn)
```
Comment (youngjeong46) [inline lib/kasten-k10-construct/index.ts:9]: also, does the addon require any Secrets? Only thing I am aware of is data encryption for object/nfs file, but want to ensure that the patterns don't break due to missing secrets.

**Comment by shapirov103** [comment]
this will be closed in favor of #27

**Pull request closed without merging**